pkgs/applications/networking/cluster/argocd-cli/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib, buildGoPackage, fetchFromGitHub, packr }:
+
+buildGoPackage rec {
+  pname = "argocd-cli";
+  version = "1.3.0";
+
+  goPackagePath = "github.com/argoproj/argo-cd";
+
+  goDeps = ./deps.nix;
+
+  src = fetchFromGitHub {
+    owner = "argoproj";
+    repo = "argo-cd";
+    rev = "v${version}";
+    sha256 = "0g3nfrdls61wqa2p2xp9vlbqlm7kk4llbzqqydk3vkr99i2whvml";
+  };
+ 
+  nativeBuildInputs = [ packr ];
+ 
+  buildPhase = ''
+    # remove nested dependencies
+    rm -rf go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery
+    rm -rf go/src/k8s.io/kubernetes/vendor/k8s.io/api
+
+    cd go/src/${goPackagePath}
+    packr
+    go build cmd/argocd/main.go
+  '';
+
+  installPhase = ''
+    install -D main $bin/bin/argocd
+  '';
```

Comment:
Are you trying to pack only one binary? You can use the `subPackages` list of strings instead, such as `subPackages = [ "cmd/argocd" ];`. However, what about other binaries? Shouldn't we include all of them?

Reply:
Added all binaries specified in make file.